### DISCUSSION TITLE
Image Update automation Error  ' remote authentication required but no callback set '

### DISCUSSION THREAD

**Discussion by JeanYvesQueignec** [fluxcd/flux2#1656] [Q&A] [answered]
Hi,
When I use Flux-system as Gitrepository as sourceRef in my fluxsystem-automation.yaml file I do not get issue 
But when I set an other repository as sourceRef  (which is also placed on the same Gitlab account- so with the right access token) I get the  error message " remote authentication required but no callback set"
Any idea where this issue comes from ?
Regards,
Jean-Yves

**Comment by jwerre** [comment]
Did you set the correct `sourceRef`? You should be able to find it like this: `kubectl get gitrepository flux-system -n flux-system -o jsonpath={.spec.secretRef.name}` assuming you're using the `flux-system` namespace.

**Reply by JeanYvesQueignec** [reply]
Hi , thanks for your quick reply. Actually , I have referenced the repository hw for Flux with :
flux create source git hw --url=https://gitlab.com/.../hw --branch=main 
 generating GitRepository source
► applying GitRepository source
✔ GitRepository source updated
◎ waiting for GitRepository source reconciliation
✔ GitRepository source reconciliation completed
✔ fetched revision: main/4e97dd5ba53ab29d8cc011cc72b00f3451e8cdad

So when I do
kubectl get gitrepository hw -n flux-system -o jsonpath={.spec.secretRef.name} I get nothing
whereas I get flux-system  for  kubectl get gitrepository flux-system -n flux-system -o jsonpath={.spec.secretRef.name}

**Reply by jwerre** [reply]
right, but if you remove the `-o` flag you should see something like this correct?

```
kubectl get gitrepository flux-system -n flux-system
NAME             URL                                                   READY   STATUS                                                              AGE
flux-system    https://gitlab.com/.../hw                  True       ...
```

**Reply by JeanYvesQueignec** [reply]
OK So you must link flux-system to the repo https://gitlab.com/.../hw  ??
You can not set hw instead 
NAME             URL                                                   READY   STATUS                                                              AGE
hw             https://gitlab.com/.../hw                  True       ...

and keep 
NAME             URL
flux-system     https://gitlab.com/.../flux-system

**Reply by jwerre** [reply]
Honestly, I'm a bit confused about this as well. From what I can understand I believe your  `GitRepository` should be in the `flux-system` namespace. As well as your `ImageUpdateAutomation`, `ImageRepository` and `ImagePolicy`.

**Reply by JeanYvesQueignec** [reply]
Thanks for these information @jwerre , I am also confused by this configuration as Flux was supposed to follow the repositories  provided as sources.

**Reply by JeanYvesQueignec** [reply]
Between I have checked on other examples in Github and did not see any other sourceRef than flux-system.
So I presume automation can only concern YAML deployment files set on the Flux repo

**Comment by JeanYvesQueignec** [comment, marked as answer]
Hello ,
To use  an other repository, it is necessary to create a secret 

flux create secret git hw-secret --url=https://gitlab.com/$GITLAB_USER/hw\
 --username=$GITLAB_USER --password=****

and  refer to it in the source:
apiVersion: source.toolkit.fluxcd.io/v1beta1
kind: GitRepository
metadata:
  name: hw
  namespace: flux-system
spec:
  interval: 30s
  ref:
    branch: main
  url: https://gitlab.com/$GITLAB_USER/hw.git
  secretRef:
    name: hw-secret
```
